pandas/_libs/tslib.pyx

```diff
@@ -2622,8 +2622,6 @@ cdef class _Timedelta(timedelta):
             int ndim
 
         if isinstance(other, _Timedelta):
-            if isinstance(other, _NaT):
```

Comment:
why do you think this can be removed? are the NaT comparisons handled higher up (IOW they never hit this)? (I don't remember)

Reply:
See discussion in #17379.  TLDR: `_NaT` does not subclass `_Timedelta`, so something very weird would have to be going on for this condition to be satisfied.

Reply:
```
In [1]: pd.Timedelta('1 day') > pd.NaT
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-9550edbbc777> in <module>()
----> 1 pd.Timedelta('1 day') > pd.NaT

/Users/jreback/pandas/pandas/_libs/tslib.pyx in pandas._libs.tslib._Timedelta.__richcmp__ (pandas/_libs/tslib.c:47402)()
   2653                 elif op == Py_NE:
   2654                     return True
-> 2655                 raise TypeError('Cannot compare type %r with type %r' %
   2656                                 (type(self).__name__, type(other).__name__))
   2657 

TypeError: Cannot compare type 'Timedelta' with type 'NaTType'

In [2]: pd.Timedelta('1 day') == pd.NaT
Out[2]: False

```

right so this is ok